### PR TITLE
Generalised avatar view and change flow

### PR DESCRIPTION
**Pull request by kumarpalsinh25**
Make avatar view and change flow generic and cover different use-case as below.

### 1) Avatar Empty - With Change Permission - Direct change options:

https://github.com/user-attachments/assets/ffb1f842-eafb-4489-8746-1e5526269753

### 2) Avatar Available - With Change Permission - Full View + Change:

https://github.com/user-attachments/assets/65046f1d-490e-427b-847b-417f3f73e7d0

### 3) Avatar Available - No Permission - Full View Only:

https://github.com/user-attachments/assets/980847c1-1c2a-48c1-be81-c639d5769948

### 4) Avatar Empty - No Permission - No Full screen View:

https://github.com/user-attachments/assets/a88a1b09-92f2-4de3-a751-27719cb95e5f

### 5) Zoom option:

https://github.com/user-attachments/assets/16ef586f-0d38-41c0-aa04-06f42191e53f

Side-note:
- 
Full screen avatar is still not support for DM and Group as it need to manage with different scenario.

### CODE REVIEW

File: app/lib/common/utils/utils.dart
Comment (gnunicorn) [inline app/lib/common/utils/utils.dart:197]: always using `ref.read` in function calls, as we are not actually setting up any watcher and data isn't being executed if things changed. Using `read` clarifies that this is the case.

```suggestion
  final membership = await ref.read(roomMembershipProvider(roomId).future);
  final canUpdateAvatar = membership?.canString('CanUpdateAvatar') == true;
  final avatarInfo = ref.read(roomAvatarInfoProvider(roomId));
```

File: app/lib/common/utils/utils.dart
Comment (gnunicorn) [inline app/lib/common/utils/utils.dart:218]: :+1:

File: app/lib/common/utils/utils.dart
Comment (gnunicorn) [inline app/lib/common/utils/utils.dart:233]: `EasyLoading.showError` please. Tell the user if we couldn't complete it or they assume it worked and well, it didn't ;) .

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (gnunicorn) [inline app/lib/common/widgets/avatar/full_screen_avatar_page.dart:22]:
```suggestion
    final membership = ref.watch(roomMembershipProvider(roomId)).valueOrNull;
```

I feel like you should create a custom rewrite in your editor that replace `requireValue` with `valueOrNull` on typing ;) .

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (gnunicorn) [inline app/lib/common/widgets/avatar/full_screen_avatar_page.dart:33]: When seeing the videos I was wondering about resetting the avatar ... like, just removing it... but we don't even have the rust API ready for that yet, so ... nothing here for now ;) .

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (gnunicorn) [inline app/lib/common/widgets/avatar/full_screen_avatar_page.dart:32]: Should we use the upload icon here? the pencil implies I _edit the image itself_, rather than replace it...

**Review by gnunicorn — APPROVED** [review]
nice work. some minor requireValue stuff that needs to be fixed first ;)

**Review by gtalha07 — APPROVED** [review]
The code looks fine to me, as well as the new UI improvements. But I have few suggestions for improved user-experience in regards of design flow. Specifically I see (from screen videos), when proceeding direct-change avatar flow, there should be confirmation sheet before opening picker which can show options .i.e. ```Change/Upload Avatar``` and ```Cancel```. This is good extra step for user and provides more secure and intended way of changing important data. Overall, this goes as my approval but if it's something trivial that can be added right away before merge, I would encourage but I'd leave that up to you!

**Comment by gnunicorn** [comment]
>  Specifically I see (from screen videos), when proceeding direct-change avatar flow, there should be confirmation sheet before opening picker which can show options .i.e. Change/Upload Avatar and Cancel. This is good extra step for user and provides more secure and intended way of changing important data. 

Why? If you didn't intend to do that, you just cancel/navigate back... I don't think this extra step is helpful unless there was an actual "option" here ... e.g. "Select File", "Take Picture with Camera" . Otherwise it is clicked wrong once but learnt immediately through that. I'd not do that extra step, as it is also not harmful in any way.

**Comment by gtalha07** [comment]
> > Specifically I see (from screen videos), when proceeding direct-change avatar flow, there should be confirmation sheet before opening picker which can show options .i.e. Change/Upload Avatar and Cancel. This is good extra step for user and provides more secure and intended way of changing important data.
> 
> Why? If you didn't intend to do that, you just cancel/navigate back... I don't think this extra step is helpful unless there was an actual "option" here ... e.g. "Select File", "Take Picture with Camera" . Otherwise it is clicked wrong once but learnt immediately through that. I'd not do that extra step, as it is also not harmful in any way.

Well, I didn't mean to put in harmful sense, but that way, it could be more informational about what this interaction actually means. Because we have different actions on avatars currently, i.e. tapping on parent badge leads to space but tapping on the navigated space avatar opens up picker. Unless we plan on documenting this somewhere else?? Anyways just my thoughts...

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (kumarpalsinh25) [reply app/lib/common/widgets/avatar/full_screen_avatar_page.dart:22]: Oh no, Again!! :(

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (kumarpalsinh25) [reply app/lib/common/widgets/avatar/full_screen_avatar_page.dart:33]: I also thought on the same during this PR development, but kept it for later.

File: app/lib/common/widgets/avatar/full_screen_avatar_page.dart
Comment (kumarpalsinh25) [reply app/lib/common/widgets/avatar/full_screen_avatar_page.dart:32]: Yes sure

**Pull request merged**
